custom-configs/src/main/java/com/sequenceiq/cloudbreak/service/CustomConfigsService.java

```diff
@@ -0,0 +1,219 @@
+package com.sequenceiq.cloudbreak.service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.cloudera.api.swagger.model.ApiClusterTemplateRoleConfigGroup;
+import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+import com.sequenceiq.authorization.service.ResourcePropertyProvider;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.dto.NameOrCrn;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.crn.Crn;
+import com.sequenceiq.cloudbreak.auth.crn.CrnResourceDescriptor;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareCrnGenerator;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.cloudbreak.domain.CustomConfigProperty;
+import com.sequenceiq.cloudbreak.domain.CustomConfigs;
+import com.sequenceiq.cloudbreak.exception.CustomConfigsException;
+import com.sequenceiq.cloudbreak.repository.CustomConfigsRepository;
+
+@Service
+public class CustomConfigsService implements ResourcePropertyProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomConfigsService.class);
+
+    @Inject
+    private CustomConfigsRepository customConfigsRepository;
+
+    @Inject
+    private CustomConfigsValidator validator;
+
+    @Inject
+    private RegionAwareCrnGenerator regionAwareCrnGenerator;
+
+    public void initializeCrnForCustomConfigs(CustomConfigs customConfigs, String accountId) {
+        customConfigs.setResourceCrn(regionAwareCrnGenerator.generateCrnStringWithUuid(CrnResourceDescriptor.CUSTOM_CONFIGS, accountId));
+    }
+
+    public List<CustomConfigs> getAll(String accountId) {
+        return customConfigsRepository.findCustomConfigsByAccountId(accountId);
+    }
+
+    public CustomConfigs getByNameOrCrn(NameOrCrn nameOrCrn, String accountId) {
+        return nameOrCrn.hasName() ? getByName(nameOrCrn.getName(), accountId) : getByCrn(nameOrCrn.getCrn());
+    }
+
+    public CustomConfigs getByCrn(String crn) {
+        CustomConfigs toReturn = null;
```

Comment:
make it protected or private since getByNameOrCrn is there. crn cannot be null here (Endpoint api should have not null constraint for name or crn inputs)  null check can be removed and a single return statement can be used.